sweep/sweeper.go

```diff
@@ -395,7 +395,7 @@ func New(cfg *UtxoSweeperConfig) *UtxoSweeper {
 		pendingSweepsReqs: make(chan *pendingSweepsReq),
 		quit:              make(chan struct{}),
 		inputs:            make(InputsMap),
-		bumpResultChan:    make(chan *BumpResult, 100),
+		bumpRespChan:      make(chan *bumpResp, 100),
```

Comment:
could you explain why we buffer here and not just go with an unbuffered channel ? According to my understanding we already launch goroutines for the `monitorFeeBumpResult` so I think it's ok for them to just wait or ?

And we could move the `CancelRebroadcast` before the channel operation so we make sure we cancel it early ?

Reply:
`bumpRespChan` is used by all sweeping txns, so even in goroutine if the callsite is slow then it may get stuck.

> And we could move the CancelRebroadcast before the channel operation so we make sure we cancel it early ?

Not sure I follow here, also think this is out of scope tho.

Reply:
yes it is out of scope, I tried to understand the reasoning behind this for my future code changes, so basically we should always try to write into a buffer even if we can afford blocking on the individual goroutine ?

Reply:
ohh now I understand there are even more dependencies, but the most important one is, that we do not handle the the initial broadcast error in a goroutine, so if we had no buffer, this could deadlock our sweeper:

```
func (t *TxPublisher) Broadcast(req *BumpRequest) <-chan *BumpResult {
	log.Tracef("Received broadcast request: %s",
		lnutils.SpewLogClosure(req))

	// Store the request.
	requestID, record := t.storeInitialRecord(req)

	// Create a chan to send the result to the caller.
	subscriber := make(chan *BumpResult, 1)
	t.subscriberChans.Store(requestID, subscriber)

	// Publish the tx immediately if specified.
	if req.Immediate {
		t.handleInitialBroadcast(record, requestID)
	}

	return subscriber
}
```

so there needs to be a buffer, but I would even prefer handling the `handleInitialBroadcast` in a. separate goroutine (but not sure)

Reply:
Nit: I think it would be great to have a good explanation why we have a buffer of size 100.

Reply:
> that we do not handle the the initial broadcast error in a goroutine, so if we had no buffer, this could deadlock our sweeper:

I think no? Why would this block the sweeper?

Reply:
Flow:

Ohh right I thought Broadcast would not write in a buffered channel, but now this makes sure that `handleInitialBroadcast` never blocks. So my conclusion is not correct, so acutally I think we don't really need this global buffer because the monitoring goroutines (`go s.monitorFeeBumpResult(set, resp)`) would be good enough but maybe the buffered channels has really some performance benefits as you mentioned.